quickwit/quickwit-search/src/collector.rs

```diff
@@ -154,25 +154,53 @@ impl SortingFieldExtractorComponent {
     ///
     /// The function returns None if the sort key is a fast field, for which we have no value
     /// for the given doc_id, or we sort by DocId.
+    #[inline]
     fn extract_typed_sort_value_opt(&self, doc_id: DocId, score: Score) -> Option<SortValue> {
+        match self {
+            SortingFieldExtractorComponent::DocId => None,
+            SortingFieldExtractorComponent::FastField {
+                sort_column,
+                sort_field_type: _,
+                ..
+            } => sort_column.first(doc_id).map(SortValue::U64),
+            SortingFieldExtractorComponent::Score { .. } => Some(SortValue::F64(score as f64)),
+        }
+    }
+
+    #[inline]
+    /// Converts u64 fast field values to its correct type.
+    /// The conversion is delayed for performance reasons.
+    fn convert_u64_ff_val_to_sort_value(&self, sort_value: SortValue) -> SortValue {
         let map_fast_field_to_value = |fast_field_value, field_type| match field_type {
             SortFieldType::U64 => SortValue::U64(fast_field_value),
             SortFieldType::I64 => SortValue::I64(i64::from_u64(fast_field_value)),
             SortFieldType::F64 => SortValue::F64(f64::from_u64(fast_field_value)),
             SortFieldType::DateTime => SortValue::I64(i64::from_u64(fast_field_value)),
             SortFieldType::Bool => SortValue::Boolean(fast_field_value != 0u64),
         };
-
         match self {
-            SortingFieldExtractorComponent::DocId => None,
+            SortingFieldExtractorComponent::DocId => sort_value,
             SortingFieldExtractorComponent::FastField {
-                sort_column,
-                sort_field_type,
-                ..
-            } => sort_column
-                .first(doc_id)
-                .map(|field_val| map_fast_field_to_value(field_val, *sort_field_type)),
-            SortingFieldExtractorComponent::Score { .. } => Some(SortValue::F64(score as f64)),
+                sort_field_type, ..
+            } => match sort_value {
+                SortValue::U64(val) => map_fast_field_to_value(val, *sort_field_type),
+                _ => panic!("Internal error: Got non-U64 sort value for fast field."),
+            },
+            SortingFieldExtractorComponent::Score => sort_value,
+        }
+    }
+    #[inline]
+    /// Converts fast field values into their u64 fast field representation.
+    fn convert_to_u64_ff_val(&self, sort_value: SortValue) -> SortValue {
+        match self {
+            SortingFieldExtractorComponent::DocId => sort_value,
+            SortingFieldExtractorComponent::FastField { .. } => match sort_value {
+                SortValue::U64(_val) => sort_value,
```

Comment:
I think there is a bug here, aren't there?

The partial hit could be coming from a different split, where the value was i64 for instance.
The current split on the other hand could be `u64`.

In that case I believe the ordering is broken.

Reply:
Indeed, this is broken currently. 
This needs to reflect `impl Ord for SortValue`.


This comparison on `SortValue` is also something that I couldn't attribute in a profile:
6% time spent in `quickwit_proto::search::<...>::cmp`  in the
quickwit generated log benchmark on query `sort_numeric_desc.json`:
```json
{
  "track_total_hits": true,
  "query": {
    "match_all": {}
  },
  "sort": [
    {
      "metrics.size": {"order": "desc"}
    }
  ]
}
```